### PR TITLE
Add wasi support

### PR DESCRIPTION
**Pull request by quink-black**
WASI is WebAssembly System Interface. Now we can build ncnn with wasi-sdk[1] to create a WebAssembly binary module, and run it with wasmtime on cmdline.

[1] https://github.com/WebAssembly/wasi-sdk
[2] https://github.com/bytecodealliance/wasmtime

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=5534) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=5534) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=5534) it.</sub>

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -395,9 +395,10 @@ if(NCNN_TARGET_ARCH STREQUAL "x86")
             endif()
             target_compile_options(ncnn PRIVATE /D__SSE2__)
         else()
-            target_compile_options(ncnn PRIVATE -msse2 -msse)
-            if(CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
+            if(CMAKE_SYSTEM_NAME MATCHES "Emscripten|WASI")
                 target_compile_options(ncnn PRIVATE -msimd128)
+            else()
+                target_compile_options(ncnn PRIVATE -msse2 -msse)
```
Comment (quink-black) [inline src/CMakeLists.txt:401]: I'm not sure whether "-msse2 -msse" should keep for Emscripten.

File: src/benchmark.cpp [hunk 2]
```diff
@@ -15,8 +15,18 @@
...
+#else
+#define USE_CXX11_CLOCK 0
+#endif
+
+#if USE_CXX11_CLOCK
 #include <chrono>
+#if NCNN_THREADS
 #include <thread>
+#else
+#include <unistd.h>   // usleep()
```
Comment (quink-black) [inline src/benchmark.cpp]: Ok, I made another mistake.

**Comment by nihui** [comment]
Thanks for your contribution !

**Pull request merged**